VizRank-like function for Box Plot

janezd opened the pull request:
<del>Based on #1405 (not master).</del>

ajdapretnar commented:
@janezd Is this still WIP?

janezd commented:
Blocked by #1405.

ajdapretnar commented:
Ah, yes, of course. :panda_face:

kernc commented on Orange/widgets/visualize/owboxplot.py:
```diff
@@ -224,14 +232,56 @@ def set_data(self, dataset):
+                groups = (col[group_col == i] for i in range(n_groups))
+                groups = (col[~np.isnan(col)] for col in groups)
+                groups = [group for group in groups if len(group)]
+                return f_oneway(*groups)[1] if groups else 2
```
To wet the appetite, in pandas, this could be:
```py
if attr.is_continuous:
    groups = data.groupby(group_var)[attr].apply(list)
    f_oneway(*groups)[1] if len(groups) else 2

janezd replied:
Interestingly, it's *whet*, not *wet*.

VesnaT commented:
When attribute's score appears to be nan (i.e not all values of discrete variable appear in a subset, which is then sent to the Boxplot or there is only one distinct value for a continuous variable in the subset), variables doesn't seem to get sorted correctly. I would put those (with nan score) at the end of the list.

janezd commented:
I added commit 9be832d945c77f320eb23e3fa96af7306883916a. Does it fix the problem? For discrete variables, it can still compute the p-value if there are at least two distinct values. Otherwise, it puts the attribute at the end, as you suggested. I think I also fixed the problem with constant numeric values.

If it still doesn't work: can you send me the data (or instructions) for reproducing this?

VesnaT commented:
Still not working for continuous variables. What about returning 2 when the score is nan?

janezd commented:
Thanks. Please try now.
